analyzers/src/SonarAnalyzer.CSharp/Rules/RedundantDeclaration.cs

```diff
@@ -149,49 +141,38 @@ private static bool ParameterTypesDoNotMatch(IMethodSymbol method1, IMethodSymbo
 
         private static void ReportRedundantNullableConstructorCall(SyntaxNodeAnalysisContext context)
         {
-            var objectCreation = (ObjectCreationExpressionSyntax)context.Node;
+            var objectCreation = ObjectCreationFactory.Create(context.Node);
             if (!IsNullableCreation(objectCreation, context.SemanticModel))
             {
                 return;
             }
 
-            if (IsInNotVarDeclaration(objectCreation) ||
-                IsInAssignmentOrReturnValue(objectCreation) ||
-                IsInArgumentAndCanBeChanged(objectCreation, context.SemanticModel))
+            if (IsInNotVarDeclaration(objectCreation.Expression)
+                || IsInAssignmentOrReturnValue(objectCreation.Expression)
+                || IsInArgumentAndCanBeChanged(objectCreation, context.SemanticModel))
             {
-                ReportIssueOnRedundantObjectCreation(context, objectCreation, "explicit nullable type creation", RedundancyType.ExplicitNullable);
+                ReportIssueOnRedundantObjectCreation(context, objectCreation.Expression, "explicit nullable type creation", RedundancyType.ExplicitNullable);
             }
         }
 
-        private static bool IsNullableCreation(ObjectCreationExpressionSyntax objectCreation, SemanticModel semanticModel)
-        {
-            if (objectCreation.ArgumentList == null ||
-                objectCreation.ArgumentList.Arguments.Count != 1)
-            {
-                return false;
-            }
+        private static bool IsNullableCreation(IObjectCreation objectCreation, SemanticModel semanticModel) =>
+            objectCreation.ArgumentList is {Arguments: {Count: 1}} && objectCreation.TypeSymbol(semanticModel).OriginalDefinition.Is(KnownType.System_Nullable_T);
 
-            var type = semanticModel.GetSymbolInfo(objectCreation).Symbol?.ContainingType;
-            return type != null &&
-                type.OriginalDefinition.Is(KnownType.System_Nullable_T);
-        }
-
-
-        private static bool IsInAssignmentOrReturnValue(ObjectCreationExpressionSyntax objectCreation)
-        {
-            var parent = objectCreation.GetFirstNonParenthesizedParent();
-            return parent is AssignmentExpressionSyntax ||
-                parent is ReturnStatementSyntax ||
-                parent is LambdaExpressionSyntax;
-        }
+        private static bool IsInAssignmentOrReturnValue(SyntaxNode objectCreation) =>
+            objectCreation.GetFirstNonParenthesizedParent() switch
+            {
+                AssignmentExpressionSyntax assignment => true,
+                ReturnStatementSyntax @return => true,
+                LambdaExpressionSyntax lambda => true,
```

Comment:
We don't need to name them, please use discard to avoid unused variables. We should have a rule about this (or update existing one)
```suggestion
                AssignmentExpressionSyntax _ => true,
                ReturnStatementSyntax _ => true,
                LambdaExpressionSyntax _ => true,
```

Reply:
Fixed to _. QG didn't complain about that.